include/pros/detail.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * \file pros/detail.hpp
+ *
+ * Contains PROS internal functions used in port literal functions.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2023 Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License v. 2.0. If a copy of the MPL was not distributed with this
+ * file You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ */
+
+#include <array>
+#include <charconv>
+#include <cstdint>
+
+namespace pros::detail {
+template <char ...Chars>
+consteval uint8_t parse_port() {
+    const std::array chars{Chars...};
+    const char* str = chars.begin();
+    int base = 10;
+    uint8_t port_num{0};
+    if (chars.size() > 2 && chars[0] == '0') {
+        str += 2;
+        // from_chars doesn't seem to handle hex/binary/octal prefixes,
+        // so we handle them ourselves
```

Comment:
This sucks lmao, good catch though

Reply:
Turns out it does, I failed to read the docs lol. Fixed in https://github.com/purduesigbots/pros/pull/689/commits/bea9c443195fbecb738a4154f2048603fa4dd178.

Reply:
oh thank god

Reply:
Turns out I failed to read the docs doubly lol: from_chars does indeed NOT auto detect base womp womp